### ISSUE TITLE
pitchRatio is twice the value it should be...

### ISSUE THREAD

**Issue by hottwaj** [schellingb/TinySoundFont#21]
Not sure if this is an issue specific to the soundfonts I've tried, but I've found that TSF is playing samples at circa double the speed they should be played at.  This can be quite noticeable for soundfonts where the sound changes dramatically over different octaves...

I haven't got stuck into the details, but my fix has been to scale pitchRatio by 0.5 on line #[L978](https://github.com/schellingb/TinySoundFont/blob/master/tsf.h#L978)...  Otherwise I always see pitchRatio values between ~1.5 and ~2.5, i.e. samples are being played at twice the speed they should be...

This might also apply to dynamic pitch ratio calculations - I haven't been able to check yet though...

Hope that makes sense and thanks for the great library!

**Comment by schellingb** [comment]
Hello there!

Can you post a download link to the sf2 file you are using? Or maybe can you confirm if you can hear a difference from how it sounds inside a soundfont editor like [Viena](http://www.synthfont.com/Downloads.html)?

Also, are you playing back notes with the synth API (tsf_note_on) or are you using the MIDI channel interface (tsf_channel_note_on, etc)?

I'm pretty sure this can't be the right solution to just apply a global .5 scale. My own tests and other reports I received suggest that at least in most cases the pitch seems correct. Maybe some special case?

Thanks for the report and the testing, hopefully we can figure this one out!

**Comment by hottwaj** [comment]
Oh man, actually this was all my fault!  I tried to write an isolated test which I found didn't exhibit the problem...  Then on further digging I found I had introduced a bug where I was:
- setting the pitch bend range to +/- 1 octave
- sending a set pitch wheel message
- then playing notes

But, I had got the order of the two pitch wheel bytes wrong, so everything was being pitch-bent up by 1 octave, hence the problem I had observed.

Sorry!  And thanks for the quick reply :)
